### PR TITLE
fix: multitenancy changes

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change
(A few sentences about this PR)

## Related issues
- https://github.com/supertokens/supertokens-core/pull/555
- https://github.com/supertokens/supertokens-core/issues/546

## Test Plan
(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes
(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates
- [ ] Changelog has been updated
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
   - In `build.gradle`
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR
- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java [hunk 1]
```diff
@@ -926,6 +926,22 @@ private static List<? extends AuthRecipeUserInfo> getUserInfoForRecipeIdFromUser
         }
     }
 
+    public static String getRecipeIdForUser_Transaction(Start start, Connection sqlCon, TenantIdentifier tenantIdentifier, String userId)
+            throws SQLException, StorageQueryException {
+        String QUERY = "SELECT recipe_id FROM " + getConfig(start).getUsersTable()
+                + " WHERE app_id = ? AND user_id = ? FOR UPDATE";
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java:932]: the primary key here is (app_id, tenant_id, user_id). This means that you can get multiple recipe_id -> is it guaranteed that all of them will be the same always?

File: src/main/java/io/supertokens/storage/postgresql/queries/EmailPasswordQueries.java [hunk 2]
```diff
@@ -398,6 +398,41 @@ public static UserInfo getUserInfoUsingEmail(Start start, TenantIdentifier tenan
         });
     }
 
+    public static boolean addUserIdToTenant_Transaction(Start start, Connection sqlCon,
+                                                        TenantIdentifier tenantIdentifier, String userId)
+            throws SQLException, StorageQueryException {
+        UserInfo userInfo = EmailPasswordQueries.getUserInfoUsingId_Transaction(start, sqlCon,
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/queries/EmailPasswordQueries.java:404]: why do you need to do getUserInfoUsingId_Transaction? and not just getUserInfoUsingId? Cause this is already in a transaction.

File: src/main/java/io/supertokens/storage/postgresql/queries/PasswordlessQueries.java [hunk 3]
```diff
@@ -747,6 +763,41 @@ public static UserInfo getUserByPhoneNumber(Start start, TenantIdentifier tenant
         });
     }
 
+    public static boolean addUserIdToTenant_Transaction(Start start, Connection sqlCon, TenantIdentifier tenantIdentifier, String userId)
+            throws StorageQueryException, SQLException {
+        UserInfo userInfo = PasswordlessQueries.getUserById_Transaction(start, sqlCon,
+                tenantIdentifier.toAppIdentifier(), userId);
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/queries/PasswordlessQueries.java:769]: why do you need to do getUserInfoUsingId_Transaction? and not just getUserInfoUsingId? Cause this is already in a transaction.

File: src/main/java/io/supertokens/storage/postgresql/queries/ThirdPartyQueries.java [hunk 4]
```diff
@@ -313,6 +330,40 @@ public static UserInfo[] getThirdPartyUsersByEmail(Start start, TenantIdentifier
         });
     }
 
+    public static boolean addUserIdToTenant_Transaction(Start start, Connection sqlCon, TenantIdentifier tenantIdentifier, String userId)
+            throws SQLException, StorageQueryException {
+        UserInfo userInfo = ThirdPartyQueries.getUserInfoUsingUserId_Transaction(start, sqlCon,
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/queries/ThirdPartyQueries.java:335]: why do you need to do getUserInfoUsingId_Transaction? and not just getUserInfoUsingId? Cause this is already in a transaction.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java (hunk 1 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java:932]: Yes. Initially there is only one row. While adding new rows, we add the same recipe id.

File: src/main/java/io/supertokens/storage/postgresql/queries/PasswordlessQueries.java (hunk 3 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/queries/PasswordlessQueries.java:769]: Done

File: src/main/java/io/supertokens/storage/postgresql/queries/ThirdPartyQueries.java (hunk 4 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/queries/ThirdPartyQueries.java]: Done

File: src/main/java/io/supertokens/storage/postgresql/queries/MultitenancyQueries.java [hunk 5]
```diff
@@ -220,4 +238,35 @@ public static void addTenantIdInUserPool(Start start, TenantIdentifier tenantIde
...
+                });
+            }
+
+        } catch (SQLException throwables) {
+            throw new StorageQueryException(throwables);
+        }
+    }
+
+    public static boolean removeUserIdFromTenant(Start start, TenantIdentifier tenantIdentifier, String userId) {
+        return false;
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/queries/MultitenancyQueries.java]: Done

File: src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java (hunk 1 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java]: Now querying from app_id_to_user_id table, which will contain only one row.

**Pull request merged**